src/udp.cpp

```diff
@@ -171,9 +179,15 @@ void UDP::write_serialization(uint8_t* buffer, uint32_t total_sz) {
     while (checksum >> 16) {
         checksum = (checksum & 0xffff)+(checksum >> 16);
     }
-    header_.check = ~checksum;
-    // If checksum is 0, it has to be set to 0xffff
-    header_.check = (header_.check == 0) ? 0xffff : header_.check;
+    if (use_checksum_) {
+        header_.check = ~checksum;
+        // If checksum is 0, it has to be set to 0xffff
+        header_.check = (header_.check == 0) ? 0xffff : header_.check;
+    }
+    else {
+        //Unused checksum transmitted as 0
+        header_.check = 0;
```

Comment:
This is based on the RoCEv2 spec, right? Or is there an IETF standard for not using the checksum in UDP or elsewhere?

Reply:
Okay, after vicious battles with Tins, static linking, endianness, and auto serialization ... new change is in @ebertland please review.  

This change solves the two issues I brought up in the review: 1) infinite recursion on calculating icrc on serialization, and 2) hardcoding IP/UDP checksum byte offsets.  There is now a setter function that enables setting IP/UDP checksum, and once used, the caller assumes responsibility for setting it correctly, since it will no longer be auto calculated when the PDU is serialized.